src/tests/test_protocol.py

```diff
@@ -2,25 +2,97 @@
 Tests for common protocol functions
 """
 
+import sys
 import unittest
 
-from .common import skip_python3
-
-skip_python3()
+from pybitmessage import protocol, state
 
 
 class TestProtocol(unittest.TestCase):
     """Main protocol test case"""
 
+    def test_checkIPv4Address(self):
+        """Check the results of protocol.checkIPv4Address()"""
+        token = 'HELLO'
+        # checking protocol.encodeHost()[12:]
+        self.assertEqual(  # 127.0.0.1
+            token, protocol.checkIPv4Address(b'\x7f\x00\x00\x01', token, True))
+        self.assertEqual(  # 10.42.43.1
+            token, protocol.checkIPv4Address(b'\n*+\x01', token, True))
+        self.assertEqual(  # 192.168.0.254
+            token, protocol.checkIPv4Address(b'\xc0\xa8\x00\xfe', token, True))
+        self.assertEqual(  # 172.31.255.254
+            token, protocol.checkIPv4Address(b'\xac\x1f\xff\xfe', token, True))
+        self.assertFalse(  # 8.8.8.8
+            protocol.checkIPv4Address(b'\x08\x08\x08\x08', token, True))
+
+    def test_checkIPv6Address(self):
+        """Check the results of protocol.checkIPv6Address()"""
+        test_ip = '2001:db8::ff00:42:8329'
+        self.assertEqual(
+            'test', protocol.checkIPv6Address(
+                protocol.encodeHost(test_ip), 'test'))
+        self.assertFalse(
+            protocol.checkIPv6Address(
+                protocol.encodeHost(test_ip), 'test', True))
+
     def test_check_local(self):
         """Check the logic of TCPConnection.local"""
-        from pybitmessage import protocol, state
-
         self.assertTrue(
             protocol.checkIPAddress(protocol.encodeHost('127.0.0.1'), True))
         self.assertTrue(
             protocol.checkIPAddress(protocol.encodeHost('192.168.0.1'), True))
+        self.assertTrue(
+            protocol.checkIPAddress(protocol.encodeHost('10.42.43.1'), True))
+        self.assertTrue(
+            protocol.checkIPAddress(protocol.encodeHost('172.31.255.2'), True))
+        self.assertFalse(protocol.checkIPAddress(
+            protocol.encodeHost('2001:db8::ff00:42:8329'), True))
 
+        globalhost = protocol.encodeHost('8.8.8.8')
+        self.assertFalse(protocol.checkIPAddress(globalhost, True))
+        self.assertEqual(protocol.checkIPAddress(globalhost), '8.8.8.8')
+
+    @unittest.skipIf(
+        sys.hexversion >= 0x3000000, 'this is still not working with python3')
+    def test_check_local_socks(self):
+        """The SOCKS part of the local check"""
         self.assertTrue(
             not protocol.checkSocksIP('127.0.0.1')
             or state.socksIP)
+
+    def test_network_group(self):
+        """Test various types of network groups"""
```

Comment:
This is the new `test_network_group()`

Reply:
Okay, so I guess there's no need for test_networkgroup.py? 
Should we remove it?